corehq/warehouse/models/facts.py

```diff
@@ -122,27 +116,28 @@ class ApplicationStatusFact(BaseFact, CustomSQLETLMixin):
 
     Grain: app_id, user_id
     '''
-    # TODO: Write Update SQL Query (currently there exists a placeholder)
     slug = APP_STATUS_FACT_SLUG
 
-    # TODO: Add app dimension
-    # app_dim = models.CharField(max_length=255)
+    app_dim = models.ForeignKey(ApplicationDim, on_delete=models.PROTECT)
+    app_id = models.CharField(max_length=255)
 
-    # TODO: Add domain dimension
-    # domain_dim = models.CharField(max_length=255)
+    domain_dim = models.ForeignKey(DomainDim, on_delete=models.PROTECT)
 
     user_dim = models.ForeignKey(UserDim, on_delete=models.PROTECT)
+    user_id = models.CharField(max_length=255)
```

Comment:
i saw your comment about denormalizing and including app_id and user_id but how come that was done - couldn't we just reference those when we make queries by joining to those dims?